### PR TITLE
Update roadmap.md

### PR DESCRIPTION
**Pull request by tigrannajaryan**
1. Removed completed items.
2. Removed items that are no longer considered top priority.
3. Added queued_retry deprecation as an item.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -9,33 +9,17 @@ items from this roadmap.
...
-| |
-**Observability**|
-Add pipeline metrics| |[observability.md](observability.md)
-High-level observations|Draft|[#485](https://github.com/open-telemetry/opentelemetry-collector/issues/485)
-Research: Explore ways to expose pipelines state as a graph| |[#486](https://github.com/open-telemetry/opentelemetry-collector/issues/486)
+Metrics correctness tests|In progress|[#652](https://github.com/open-telemetry/opentelemetry-collector/issues/652)
 | |
 **New Formats**|
-OTLP Receiver and Exporter| |[#480](https://github.com/open-telemetry/opentelemetry-collector/issues/480) [#479](https://github.com/open-telemetry/opentelemetry-collector/issues/479)
-Research: OTLP over HTTP 1.1| |
```
Comment (bogdandrutu) [inline docs/roadmap.md:31]: This is missing the exporter

Comment (tigrannajaryan) [reply docs/roadmap.md:31]: The "Complete OTLP support" line includes that.

Comment (tigrannajaryan) [reply docs/roadmap.md:31]: Just to make it clear: it was a research, now the research is done. Now it is part of OTLP spec. When we say "OTLP support" it means OTLP/HTTP must be also supported.

Comment (tigrannajaryan) [reply docs/roadmap.md:31]: If the text is unclear I can try to clarify.

Comment (tigrannajaryan) [reply docs/roadmap.md:31]: Clarified.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**